### PR TITLE
SDIT-937 Add controller and services for search files

### PR DESCRIPTION
**Pull request by karenmillermoj**
The majority of this PR is a lift-and-shift (with updates for opensearch) + added in the missing ExceptionHandler

### CODE REVIEW

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/config/HmppsPrisonerSearchExceptionHandler.kt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    ResponseEntity
+      .status(BAD_REQUEST)
+      .body(
+        ErrorResponse(
+          status = BAD_REQUEST,
+          userMessage = "Validation failure: ${e.message}",
+          developerMessage = e.message,
+        ),
+      ).also { log.info("Validation exception: {}", e.message) }
+
```
Comment (karenmillermoj) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/config/HmppsPrisonerSearchExceptionHandler.kt:28]: The BadRequestExceptions current thrown can probably just change to validationexceptions - a later PR

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/KeywordService.kt [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+      throw BadRequestException("Invalid keyword search  - please provide prison locations to filter by")
+    }
+  }
+
+  fun findByKeyword(keywordRequest: KeywordRequest): Page<Prisoner> {
+    log.info("Received keyword request ${gson.toJson(keywordRequest)}")
+
+    validateKeywordRequest(keywordRequest)
+    val searchSourceBuilder = createSourceBuilder(keywordRequest)
+    val searchRequest = SearchRequest(arrayOf(INDEX_ALIAS), searchSourceBuilder)
```
Comment (karenmillermoj) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/KeywordService.kt:55]: I'm gonna rename this - I now find it annoying - will change to PRISONER_INDEX or just PRISONER

Comment (petergphillips) [reply hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/KeywordService.kt:55]: prefer `PRISONER_INDEX`

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**